pennylane_lightning/core/src/simulators/lightning_qubit/measurements/ExpValFunctorsLQubit.hpp

```diff
@@ -0,0 +1,219 @@
+// Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the License);
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an AS IS BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#pragma once
+
+#include "BitUtil.hpp"
+#include "Util.hpp"
+
+/// @cond DEV
+namespace {
+using namespace Pennylane::Util;
+} // namespace
+/// @endcond
+
+namespace Pennylane::LightningQubit::Functors {
+
+template <class PrecisionT> struct getExpectationValueIdentityFunctor {
+    const std::complex<PrecisionT> *arr;
+    size_t num_qubits;
+
+    getExpectationValueIdentityFunctor(
+        const std::complex<PrecisionT> *arr_,
+        [[maybe_unused]] std::size_t num_qubits_,
+        [[maybe_unused]] const std::vector<size_t> &wires) {
+        arr = arr_;
+        num_qubits = num_qubits_;
+    }
+
+    inline void operator()(PrecisionT &expval) const {
+        size_t k;
+#if defined(_OPENMP)
+#pragma omp parallel for default(none) shared(num_qubits, arr) private(k)      \
```

Comment:
What will happen with these OpenMP threads if we have multiple concurrent executions happening within the same process?

Reply:
I assume your point is about nested parallelism? In that case, If this region is an outer parallel region, the region is executed by creating `OMP_NUM_THREADS` threads. If this is a parallel region within an outer parallel region, with nested OpenMP enabled each thread launches `OMP_NUM_THREADS` to execute the loop, otherwise the loop is executed serially by one of the threads. This behavior can be controlled and made efficient with respect to available hardware resources by using OpenMP functions such as `omp_set_nested()` and `omp_get_max_threads()`, and explicitly setting `num_threads` for the parallel region. Please advise if this is what you mean, so I could modify the implmentaiton accordingly.

Reply:
Thanks @sbohloul No need to modify, just curious about your thoughts here.